test/Tests/src/Data/Text_Spec.enso

```diff
@@ -204,6 +206,66 @@ spec =
             'a'.each builder2.append
             builder2.to_vector . should_equal ['a']
 
+        Test.specify "should check for contains using Unicode normalization" <|
+            "Hello".contains "ell" . should_be_true
+
+            "Cześć".contains 's\u{301}' . should_be_true
+            "Cześć".contains 'c\u{301}' . should_be_true
+            "Cześć".contains 'ść' . should_be_true
+            'Czes\u{301}c\u{301}'.contains 'ś' . should_be_true
+            'Czes\u{301}c\u{301}'.contains 'ć' . should_be_true
+            'Czes\u{301}c\u{301}'.contains 'ść' . should_be_true
+            "Cześć".contains 'sc' . should_be_false
+            'Czes\u{301}c\u{301}'.contains 'sc' . should_be_false
+            "Cześć".contains 's' . should_be_false
+            "Cześć".contains 'c' . should_be_false
+            'Czes\u{301}c\u{301}'.contains 's' . should_be_false
+
+            "ABC" . contains "a" . should_be_false
+            "" . contains "foo" . should_be_false
+            "abc" . contains "" . should_be_true
+            "" . contains "" . should_be_true
+            "foo foo foo" . contains "foo" . should_be_true
+
+            "Hello!".contains "lo" . should_be_true
+            "Hello!".contains "Lo" . should_be_false
+
+        Test.specify "should allow for case-insensitive contains checks" <|
+            "Hello!".contains 'LO' (Text_Matcher Case_Insensitive.new) . should_be_true
+            "FoObar" . contains "foo" (Text_Matcher Case_Insensitive.new) . should_be_true
+            "aaaIAAA" . contains "i" (Text_Matcher Case_Insensitive.new) . should_be_true
+            "Foo" . contains "bar" (Text_Matcher Case_Insensitive.new) . should_be_false
+            "Ściana" . contains "ś" (Text_Matcher Case_Insensitive.new) . should_be_true
+            "Ściana" . contains "s" (Text_Matcher Case_Insensitive.new) . should_be_false
+
+            "Straße" . contains "ss" . should_be_false
+            "Strasse" . contains "ß" . should_be_false
+            "Straße" . contains "ss" (Text_Matcher Case_Insensitive.new) . should_be_true
+            "Strasse" . contains "ß" (Text_Matcher Case_Insensitive.new) . should_be_true
```

Comment:
Documenting this slightly peculiar case - due to how we candle case insensitive operations (tolower+toupper), and given the fact that the uppercase variant of `ß` is `SS`, `ß` and `ss` get collated in case insensitive mode.

More generally (also shown in tests here, just different place), currently in Enso: `"ß".equals_ignore_case "ss" == false`.

Not sure if this is good or bad:
1. It seems bad, because the difference that got collated is not exactly a case difference.
2. OTOH, it seems natural that these two symbols mean the same thing so under a less strict equality they may be equated.

However, in Java `"ß".equalsIgnoreCase("ss") == false`.

Moreover, it's really a different kind of difference - scharfes S is more like a ligature, i.e. in the similar spirit maybe `æ` should also get collated with `ae` etc.

So I'd lean more into the direction of trying to get rid of this collation - but I'm not exactly sure how to do this efficiently - the ICU normalizer we use for `equals_ignore_case` supports case folding, but does not accept a locale. Seems the only way to handle cases with locale is through the `to_lower_case` and `to_upper_case` methods. Interestingly, how does Java get away with this? Because their `equalsIgnoreCase` processes character-by-character (not even by grapheme clsuters!) and since the proper upper-case of `ß` is `SS` which takes two characters, the `Character.toUpperCase` simply ignores it and returns back `ß` (because it is incapable of returning two characters). So Java gets this right, because it is handling characters on a more lower lever which is too limited to encounter this issue.

Quick solutions that come to mind:
1. Use the ICU's case folding that is not locale aware, possibly adding an `if` for the Turkish locale which seems to be toggleable in ICU (maybe that's the only difference between all locales so we don't need others?).
2. Use BreakIterator and implement this manually.

(2) is likely going to be slower, so probably don't want this (although may need a benchmark to be sure). (1) could be incorrect which would be bad, unless really the only Locale having different case handling is Turkish - that's possible - but we'd need to research that - possibly ask some linguist.

I think it may make most sense to create a separate task to explore this, especially check if (1) is viable as it would be our best shot. For now I'd just live with this collation - but open to discussion if this should be resolved before merging this.

Reply:
Turns out Swifts `caseInsensitiveCompare` also compares scharfes s as equal to `ss`, so I guess we can keep the current behaviour for now.

Would probably need someone knowing German linguistics very well to understand if collating these two in case-insensitive mode makes sense or not.